src/basic/meson.build

```diff
@@ -310,8 +310,8 @@ basic_compress_sources = files(
 libbasic_compress = static_library(
         'basic-compress',
         basic_compress_sources,
-        include_directories : basic_includes,
-        dependencies : [liblz4,
+        include_directories : includes,
+        dependencies : [libdl,
                         libxz,
                         libzstd,
                         userspace],
```

Comment:
hmm, so we haven't been very good with this so far, but we should really use the "partial dependency" stuff in meson more, so that the cflags are properly added as dependencies at least.

i.e. see how libxenctrl/libiptc/libp11kit is handled in the main meson.build file. I think we should do the same for *all* libraries we dlopen().

This would both be more correct, and ensures that if people install their devel headers into some weird place we'll still function correctly, as we make sure the right include dirs are added at the right place.

In #31131 I also did it that way.